src/bm_sim/parser.cpp

```diff
@@ -465,9 +465,8 @@ ParserOpAdvance<Data>::operator()(Packet *pkt, const char *data,
   // The JSON should never specify a negative constant as the shift amount.
   assert(shift_bits.sign() >= 0);
   const auto shift_bits_uint = shift_bits.get<size_t>();
-  // TODO(antonin): temporary limitation?
-  _BM_ASSERT(shift_bits_uint % 8 == 0 &&
-             "advance bits amount must be a multiple of 8");
+  if ((shift_bits_uint % 8) != 0)
```

Comment:
Could we log an error here too?
I also counted 2 other instances of that check in the file.

Reply:
done